### PR TITLE
Add states for Loading and No Results

### PR DESCRIPTION
**Pull request by josephschmitt**
Also moves the loading of the remote data in the examples to happen inside the source function so that it can demo the loading state.

### CODE REVIEW

File: src/angularComponent/ngcModifySuggestionsTemplateFactory.js [hunk 1]
```diff
@@ -20,13 +20,30 @@ export default function ngcModifySuggestionsTemplateFactory($document, $template
...
 
     element.setAttribute('role', 'listbox');
 
+    if (loadingEl) {
+      loadingEl.setAttribute('role', 'progressbar');
+      loadingEl.setAttribute('ng-show', 'omnibox.showLoadingElement');
+    }
+
+    if (noResultsEl) {
+      noResultsEl.setAttribute('ng-show', '!omnibox.hasSuggestions() && !omnibox.isLoading');
```
Comment (onlywei) [inline src/angularComponent/ngcModifySuggestionsTemplateFactory.js]: maybe we should use `ng-if`?

File: src/angularComponent/ngcOmniboxController.js [hunk 2]
```diff
@@ -171,10 +170,22 @@ export default class NgcOmniboxController {
...
     this.highlightNone();
+    this._showLoading();
+
+    const promise = this.source({query: this.query, suggestions: this.suggestions});
+    this._sourceFunctionPromise = promise;
 
-    this.source({query: this.query, suggestions: this.suggestions}).then((suggestions) => {
-      if (suggestions) {
+    promise.then((suggestions) => {
+      // Bail out if the promise has changed
```
Comment (onlywei) [inline src/angularComponent/ngcOmniboxController.js]: I love it

File: src/angularComponent/ngcModifySuggestionsTemplateFactory.js (hunk 1 above)
Comment (josephschmitt) [inline src/angularComponent/ngcModifySuggestionsTemplateFactory.js]: ¯\_(ツ)_/¯ sure. I never really know which is preferable.

Comment (josephschmitt) [inline src/angularComponent/ngcModifySuggestionsTemplateFactory.js]: Done.

File: src/angularComponent/ngcOmniboxController.js (hunk 2 above)
Comment (josephschmitt) [inline src/angularComponent/ngcOmniboxController.js]: I aim to please.

**Pull request merged**